src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/relationships-property/edit-relationship.component.ts/edit-relationships.component.ts

```diff
@@ -0,0 +1,113 @@
+import { RelationshipService } from '@portlets/content-types/fields/service/relationship.service';
+
+interface CardinalitySorted {
+    [id: number]: RelationshipCardinality;
+}
+@Component({
+    providers: [PaginatorService],
+    selector: 'dot-edit-relationships',
+    templateUrl: './edit-relationships.component.html'
+})
+export class EditRelationshipsComponent implements OnInit {
+
+    @Output()
+    change: EventEmitter<any> = new EventEmitter();
+
+    relationshipsCurrentPage: Observable<{label: string, relationship: Relationship}[]>;
+
+    i18nMessages: {
+        [key: string]: string;
+    } = {};
+
+    private cardinalities: CardinalitySorted;
+
+    constructor(
+        public dotMessageService: DotMessageService,
+        public relationshipPaginatorService: PaginatorService,
+        private editContentTypeCacheService: EditContentTypeCacheService,
+        private relationshipService: RelationshipService) {
+
+    }
+    ngOnInit() {
+        this.dotMessageService
+            .getMessages([
+                'contenttypes.field.properties.relationship.existing.label',
+                'contenttypes.field.properties.relationship.existing.placeholder',
+            ])
+            .subscribe((res) => {
+                this.i18nMessages = res;
+            });
+
+        this.relationshipPaginatorService.url = 'v1/relationships';
+    }
+
+    /**
+     * Call when the relationship global serach changed
+     * @param any filter
+     * @memberof CategoriesPropertyComponent
+     */
+    handleRelationshipFilterChange(filter): void {
+        this.getRelationshipList(filter);
+    }
+
+    /**
+     * Call when the current page changed
+     * @param any event
+     * @memberof CategoriesPropertyComponent
+     */
+    handleRelationshipPageChange(event): void {
+        this.getRelationshipList(event.filter, event.first);
+    }
+
+    tiggerChanged(relationship: Relationship): void {
+        this.change.next({
+            velocityVar: relationship.relationTypeValue,
+            cardinality: relationship.cardinality
+        });
+    }
+
+    private getCardinalities(): Observable<CardinalitySorted> {
+        if (!this.cardinalities) {
+            this.cardinalities = {};
+
+            return this.relationshipService.loadCardinalities().pipe(
+                map((relationshipCardinalities: RelationshipCardinality[]) => {
+                    relationshipCardinalities.forEach((cardinality: RelationshipCardinality) => {
+                        this.cardinalities[cardinality.id] = cardinality;
+                    });
+
+                    return this.cardinalities;
+                })
+            );
+        } else {
+            return observableOf(this.cardinalities);
+        }
+    }
+
+    private getRelationshipList(filter = '', offset = 0): void {
+        this.relationshipPaginatorService.setExtraParams('contentTypeId', this.editContentTypeCacheService.contentType.id);
```

Comment:
Why didn't we use the original pagination service for all this? looks like is similar.